shell.nix

```diff
@@ -1,67 +1,106 @@
-{ pkgs ? import <nixpkgs> {} }: with pkgs;
+{ pkgs ? import ./pinned.nix {} }:
```

Comment:
This is possibly entirely stylistic, but I usually find that adding `with pkgs` at the beginning can make it a little confusing to track where certain things are coming from in the expression.